src/lib/factories/WorkspaceFactory.ts

```diff
@@ -0,0 +1,17 @@
+import globby from 'globby';
+import {CodeAnalyzer, Workspace} from '@salesforce/code-analyzer-core';
+
+export class WorkspaceFactory {
```

Comment:
Typically I'd do the interface/implementation pattern, but I can't think of a scenario where you'd actually want to replace this behavior, even in a test. So to limit the scope of changes, I'm just declaring the class. If in the future we want to add another implementation, it's trivial to convert to the interface/implementation pattern.

Reply:
Interfaces are for boundary layers. They help remove swap out implementations to help remove testing our dependencies. So your decision makes sense so as long we don't need to hardcode behavior for testing.